### PR TITLE
finished chapter 1

### PR DESCRIPTION
**Pull request by red-avtovo**
### Solutions for Chapter 1

Nice introduction into the language

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -417,6 +417,7 @@ task is to specify the type of this function.
 49
 -}
 
+squareSum :: Num a => a -> a -> a
```
Comment (vrom911) [inline src/Chapter1.hs:420]: We explain `Num` in the following chapters, so you are one step ahead 😸

In here `Int -> Int -> Int` would also work, as a more specific type 🙂

File: src/Chapter1.hs [hunk 2]
```diff
@@ -542,7 +544,22 @@ value after "=" where the condition is true.
...
+--   | x <= y && x >= z = x
+--   | x >= y && x <= z = x
+--   | y <= x && y >= z = y
+--   | y >= x && y <= z = y
+--   | otherwise = z
+
+mid x y z
+  | x `isBetween` (y, z) = x
+  | y `isBetween` (x, z) = y
+  | otherwise = z
```
Comment (vrom911) [inline src/Chapter1.hs:561]: This is AWESOME 😍 
Really great solution 👏🏼 Nice catch, @red-avtovo 💯

**Review by vrom911 — APPROVED** [review]
Fantastic job, @red-avtovo 🚇! Really enjoyed your solutions 🎉 

This PR is ready for merge 🚢

**Comment by red-avtovo** [comment]
Thank you so much for your review and your feedback @vrom911 :heart:

**Pull request merged**